# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1233215310 -3600
#      Thu Jan 29 08:48:30 2009 +0100
# Branch release-3-0-x
# Node ID d7d30988b3eaf30553110c8118375cc033087675
# Parent  5032328e940ba191b4f122d9b18388a22571e61c
fix fileparts tests

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2009-01-29  Jaroslav Hajek  <highegg@gmail.com>
+
+	* miscellaneous/fileparts.m: Fix test for Windows.
+
 2009-01-16  Daniel J Sebald  <daniel.sebald@ieee.org>
 
 	* plot/legend.m: Fix legend order for both horizontal and
 	vertical string cell.
 
 2009-01-13  Daniel J Sebald  <daniel.sebald@ieee.org>
 
 	* set/unique.m: Fix for vertical array inputs.
diff --git a/scripts/miscellaneous/fileparts.m b/scripts/miscellaneous/fileparts.m
--- a/scripts/miscellaneous/fileparts.m
+++ b/scripts/miscellaneous/fileparts.m
@@ -22,17 +22,21 @@
 ## @var{filename}.
 ## @seealso{fullfile}
 ## @end deftypefn
 
 function [directory, name, extension, version] = fileparts (filename)
 
   if (nargin == 1)
     if (ischar (filename))
-      ds = rindex (filename, filesep);
+      if !( findstr (octave_config_info('canonical_host_type'), 'msvc')  || findstr (octave_config_info('canonical_host_type'), 'mingw') )
+        ds = rindex (filename, filesep);
+      else
+        ds = max (rindex (filename, "\\"),rindex (filename, "/"));
+      endif
       es = rindex (filename, ".");
       ## These can be the same if they are both 0 (no dir or ext).
       if (es <= ds)
 	es = length(filename)+1;
       endif
       if (ds == 0)
 	directory = "";
       elseif (ds == 1)
@@ -60,34 +64,34 @@ endfunction
 %! [d, n, e] = fileparts ("file");
 %! assert (strcmp (d, "") && strcmp (n, "file") && strcmp (e, ""));
 
 %!test
 %! [d, n, e] = fileparts ("file.ext");
 %! assert (strcmp (d, "") && strcmp (n, "file") && strcmp (e, ".ext"));
 
 %!test
-%! [d, n, e] = fileparts ("/file.ext");
-%! assert (strcmp (d, "/") && strcmp (n, "file") && strcmp (e, ".ext"));
+%! [d, n, e] = fileparts ([filesep "file.ext"]);
+%! assert (strcmp (d, filesep) && strcmp (n, "file") && strcmp (e, ".ext"));
 
 %!test
-%! [d, n, e] = fileparts ("dir/file.ext");
+%! [d, n, e] = fileparts (["dir" filesep "file.ext"]);
 %! assert (strcmp (d, "dir") && strcmp (n, "file") && strcmp (e, ".ext"));
 
 %!test
-%! [d, n, e] = fileparts ("./file.ext");
+%! [d, n, e] = fileparts (["." filesep "file.ext");
 %! assert (strcmp (d, ".") && strcmp (n, "file") && strcmp (e, ".ext"));
 
 %!test
-%! [d, n, e] = fileparts ("d1/d2/file.ext");
+%! [d, n, e] = fileparts (["d1" filesep "d2" filesep "file.ext"]);
 %! assert (strcmp (d, "d1/d2") && strcmp (n, "file") && strcmp (e, ".ext"));
 
 %!test
-%! [d, n, e] = fileparts ("/d1/d2/file.ext");
+%! [d, n, e] = fileparts ([filesep "d1" filesep "d2" filesep "file.ext"]);
 %! assert (strcmp (d, "/d1/d2") && strcmp (n, "file") && strcmp (e, ".ext"));
 
 %!test
-%! [d, n, e] = fileparts ("/.ext");
+%! [d, n, e] = fileparts ([filesep ".ext"]);
 %! assert (strcmp (d, "/") && strcmp (n, char (zeros (1, 0))) && strcmp (e, ".ext"));
 
 %!test
 %! [d, n, e] = fileparts (".ext");
 %! assert (strcmp (d, "") && strcmp (n, char (zeros (1, 0))) && strcmp (e, ".ext"));
